### PR TITLE
Added dynamic inputs and Subset Sum in C

### LINKED ISSUE #1270: Updating files to take input at runtime(#1160)
I would like to update
Subset_Sum.cpp
Subset_Sum.java
Subset_Sum.python
Subset_Sum.go

I would also like to add 
Subset_Sum.c

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #1270 referencing #1160 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Updated inputs across all languages
- Added Subset Sum in C

#### Languages Used:
- C
- C++
- Java
- Python
- Go

#### Files Added:
- Subset_Sum.c
- Subset_Sum.cpp
- Subset_Sum.py
- Subset_Sum.java
- Subset_Sum.go

### CODE REVIEW

File: Subset_Sum/Subset_Sum.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                the element in answer or leave it and take the previous
+                best given by dp[i-1][j] */
+                if (arr[i - 1] <= j)
+                    dp[i][j] = dp[i - 1][j] || dp[i - 1][j - arr[i - 1]];
+                else
+                    /* when the value of sum j is less than the element i,
+                    we cannot include this element and hence take the
+                    previous best*/
+                    dp[i][j] = dp[i - 1][j];
+             }
```
Comment (marwahmanbir) [inline Subset_Sum/Subset_Sum.c]: Fix indentation.

File: Subset_Sum/Subset_Sum.cpp [hunk 2]
```diff
@@ -32,24 +32,27 @@ bool subsetSum(ll arr[], ll size, ll sum)
...
-            best given by dp[i-1][j] */
-            if (arr[i - 1] <= j)
-                dp[i][j] = dp[i - 1][j] || dp[i - 1][j - arr[i - 1]];
-            else
-                /* when the value of sum j is less than the element i, 
-                we cannot include this element and hence take the
-                previous best*/
-                dp[i][j] = dp[i - 1][j];
+            
+            if(i > 0)
```
Comment (marwahmanbir) [inline Subset_Sum/Subset_Sum.cpp]: Add space after ``if``.

File: Subset_Sum/Subset_Sum.cpp [hunk 3]
```diff
@@ -59,16 +62,25 @@ bool subsetSum(ll arr[], ll size, ll sum)
...
-    ll arr[] = { 2, 3, 5, 1, 6, 8, 7 };
-    ll size = sizeof( arr ) / sizeof( arr[0] );
-    ll sum = 12;
+    ll size;
+    cout << "Enter size of array" << endl;
+    cin >> size;
+
+    ll arr[size];
+    cout << "Enter elements of array" << endl;
+    for(int i = 0; i < size; i++)
```
Comment (marwahmanbir) [inline Subset_Sum/Subset_Sum.cpp]: Space after ``for``.

File: Subset_Sum/Subset_Sum.java [hunk 4]
```diff
@@ -6,6 +6,8 @@
     the sum 14 by taking the elements 1, 7 and 6 
 */
 
+import java.util.Scanner;
+
 class Subset_Sum
 {
```
Comment (marwahmanbir) [inline Subset_Sum/Subset_Sum.java]: In this block, add a space after each ``for``.

File: Subset_Sum/Subset_Sum.java [hunk 5]
```diff
@@ -61,28 +63,45 @@ public static boolean subsetSum(int arr[], int size, int sum)
...
-        int arr[] = { 2, 3, 5, 1, 6, 8, 7 };
-        int size = arr.length;
-        int sum = 12;
+        Scanner sc = new Scanner(System.in);
+        System.out.println("Enter size of array");
+        int size = sc.nextInt();
+        
+        int arr[] = new int[size + 2];
+        System.out.println("Enter elements of array");
+        for(int i = 0; i < size; i++)
```
Comment (marwahmanbir) [inline Subset_Sum/Subset_Sum.java]: Space after `for`

**Comment by singh-shreya6** [comment]
@marwahmanbir I have made the requested changes. Please review.

File: Subset_Sum/Subset_Sum.java (hunk 5 above)
Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.java]: Done @marwahmanbir .

**Comment by marwahmanbir** [comment]
@singh-shreya6 there's a 13 space indentation on line 51 in C file, please check that.

**Comment by singh-shreya6** [comment]
@marwahmanbir Fixed that.

**Pull request merged**